makefiles/mcuboot.mk

```diff
@@ -1,49 +1,52 @@
+.PHONY: mcuboot mcuboot-create-key mcuboot-flash-bootloader mcuboot-flash
+
 ifdef MCUBOOT_SLOT0_SIZE
 
 IMGTOOL ?= $(RIOTTOOLS)/mcuboot/imgtool.py
-override IMGTOOL := $(abspath $(IMGTOOL))
+IMGTOOL_KEYFLAGS ?= keygen -t rsa-2048
 
-BINFILE ?= $(BINDIR)/$(APPLICATION).bin
 SIGN_BINFILE = $(BINDIR)/signed-$(APPLICATION).bin
 MCUBOOT_KEYFILE ?= $(BINDIR)/key.pem
-MCUBOOT_BIN ?= $(BINDIR)/mcuboot.bin
-MCUBOOT_BIN_URL ?= http://download.riot-os.org/mynewt.mcuboot.bin
-MCUBOOT_BIN_MD5 ?= 0c71a0589bd3709fc2d90f07a0035ce7
 
-export IMAGE_HDR_SIZE ?= 512
+MCUBOOT_ELF ?= $(ELFFILE:.elf=.mcuboot.elf)
```

Comment:
This is not a `mcuboot_elf` specific thing it is only a normal RIOT firmware but linked for `slot0`.
In the bootloader PR it is `$(BINDIR_APP)-slot0.elf`.
So maybe something like `ELFFILE_MCUBOOT_SLOT0 = $(ELFFILE:.elf=.mcuboot-slot0.elf)`